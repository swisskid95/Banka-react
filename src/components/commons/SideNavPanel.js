/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { toggleMenu } from '../../utils/sideBarOpen';

export const SideNavPanel = ({ user }) => {
  const active = { backgroundColor: '#fff', color: '#33a19d' };
  return (
    <aside className="sidebar sidebar--light">
      <span onClick={toggleMenu} className="sidebar__close">
        &times;
      </span>
      <div className="avatar">
        <img
          className="avatar__img"
          src="https://res.cloudinary.com/swisskid95/image/upload/v1566613597/Banka-assets/defaultAvatar_roxuye.png"
          alt="user"
        />
      </div>
      <h3 className="user-name">{`${user.firstname} ${user.lastname}`}</h3>
      <ul className="nav-container">
        {user.type === 'staff' ? (
          <>
            <li>
              <NavLink to="/deposit" activeStyle={active} className="nav">
                deposit
              </NavLink>
            </li>
            <li>
              <NavLink to="/withdraw" activeStyle={active} className="nav">
                withdraw
              </NavLink>
            </li>
          </>
        ) : (
          ''
        )}
        <li>
          <NavLink to="/summary" activeStyle={active} className="nav">
            summary
          </NavLink>
        </li>
        <li>
          <NavLink to="/create-account" activeStyle={active} className="nav">
            new account
          </NavLink>
        </li>
        <li>
          <Link to="/login" className="nav  logout">
            sign-out
          </Link>
        </li>
      </ul>
    </aside>
  );
};

SideNavPanel.propTypes = {
  user: PropTypes.object
};

export default SideNavPanel;
