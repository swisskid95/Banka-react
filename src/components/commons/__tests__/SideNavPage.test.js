/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { SideNavPanel } from '../SideNavPanel';

const user = {
  email: 'someuser@gmail.com',
  password: 'randompassword',
  firstname: 'some',
  lastname: 'user'
};

describe('SideNavPanel', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SideNavPanel user={user} />);

    expect(wrapper).toMatchSnapshot();
  });
});
