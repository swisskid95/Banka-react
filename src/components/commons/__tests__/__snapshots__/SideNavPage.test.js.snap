// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavPanel should render correctly 1`] = `
<aside
  className="sidebar sidebar--light"
>
  <span
    className="sidebar__close"
  >
    Ã—
  </span>
  <div
    className="avatar"
  >
    <img
      alt="user"
      className="avatar__img"
      src="https://res.cloudinary.com/swisskid95/image/upload/v1566613597/Banka-assets/defaultAvatar_roxuye.png"
    />
  </div>
  <h3
    className="user-name"
  >
    some user
  </h3>
  <ul
    className="nav-container"
  >
    <li>
      <NavLink
        activeStyle={
          Object {
            "backgroundColor": "#fff",
            "color": "#33a19d",
          }
        }
        className="nav"
        to="/summary"
      >
        summary
      </NavLink>
    </li>
    <li>
      <NavLink
        activeStyle={
          Object {
            "backgroundColor": "#fff",
            "color": "#33a19d",
          }
        }
        className="nav"
        to="/create-account"
      >
        new account
      </NavLink>
    </li>
    <li>
      <Link
        className="nav  logout"
        to="/login"
      >
        sign-out
      </Link>
    </li>
  </ul>
</aside>
`;
